print the prefix when reporting values

We should print the prefix when printing variables

so instead of 
```
npx print-env GITHUB
+ @bahmutov/print-env@2.0.1
Found environment variables:
GITHUB_ACTION=run1
GITHUB_ACTIONS=true
...
No environment variables found
##[error]Process completed with exit code 1.
```
we would show
```
npx print-env GITHUB
+ @bahmutov/print-env@2.0.1
Found environment variables that start with GITHUB:
GITHUB_ACTION=run1
GITHUB_ACTIONS=true
...
No environment variables found that start with GITHUB
##[error]Process completed with exit code 1.
```

If there are multiple prefixes, print them all
```
npx print-env FOO BAR BAZ
No environment variables found that start with FOO BAR BAZ
```

--- a/bin/print-env.js
+++ b/bin/print-env.js
@@ -21,12 +21,19 @@ const variables = getVars(prefixes)
 const keys = Object.keys(variables).sort()
 
 if (!keys.length) {
-  console.error(chalk.red('No environment variables found'))
+  console.error(chalk.red(`No environment variables found that start with ${prefixes.join(' ')}`))
   process.exit(1)
 }
 
-console.info(chalk.green.bold('Found environment variables:'))
+const found = []; const notFound = []
+
+prefixes.forEach(p => (keys.some(k => k.startsWith(p)) ? found : notFound).push(p))
+
+console.info(chalk.green.bold(`Found environment variables that start with ${found.join(' ')}:`))
 keys.forEach(key => {
   const output = `${key}${program.exists ? '' : `=${variables[key]}`}`
   console.info(output)
 })
+if (notFound.length) {
+  console.info(chalk.yellow(`No environment variables found that start with ${notFound.join(' ')}`))
+}

--- a/package.json
+++ b/package.json
@@ -63,7 +63,7 @@
     "standard": "14.3.3"
   },
   "dependencies": {
-    "chalk": "3.0.0",
+    "chalk": "4.0.0",
     "commander": "5.0.0"
   }
 }

--- a/package.json
+++ b/package.json
@@ -3,6 +3,9 @@
   "description": "Prints all environment variables that start with given string",
   "version": "0.0.0-development",
   "author": "Gleb Bahmutov <gleb.bahmutov@gmail.com>",
+  "contributors": [
+    "Jules Amosah <teomik129@gmail.com>"
+  ],
   "bugs": "https://github.com/bahmutov/print-env/issues",
   "bin": {
     "print-env": "bin/print-env.js"

--- a/src/print-env.spec.js
+++ b/src/print-env.spec.js
@@ -1,10 +1,11 @@
 'use strict'
 
 /* eslint-env mocha */
-const printEnv = require('.')
+const getVars = require('.')
 const snapshot = require('snap-shot-it')
 const execa = require('execa')
-const { join } = require('path')
+const { resolve } = require('path')
+const { strictEqual } = require('assert')
 
 describe('@bahmutov/print-env', () => {
   before(() => {
@@ -18,68 +19,43 @@ describe('@bahmutov/print-env', () => {
     })
   })
 
-  it('returns only prefixed env vars', () => {
-    snapshot(printEnv(['FOOX']))
-  })
-
-  it('supports multiple prefixes', () => {
-    snapshot(printEnv(['FOO', 'BAR']))
-  })
-
-  it('returns some found variables', () => {
-    snapshot('found at least FOO*', printEnv(['NONONO', 'FOO']))
+  it('has valid behavior for function', () => {
+    snapshot(getVars, [['FOOX']], [['FOO', 'BAR']], [['NONONO', 'FOO']])
   })
 
   describe('cli', () => {
-    const getBin = b => join(__dirname, '..', 'bin', b)
     describe('print-env', () => {
-      const bin = getBin('print-env.js')
-      it('displays usage on 0 args', async () => {
-        const { stdout } = await execa('node', [bin])
-        snapshot(stdout)
-      })
-      it('prints sorted variables with values', async () => {
-        const { stdout } = await execa('node', [bin, 'FOOX'])
-        snapshot(stdout)
-      })
-      it('prints sorted variables without values when passed -e|--exists flag', async () => {
-        const { stdout } = await execa('node', [bin, '-e', 'FOOX'])
-        snapshot(stdout)
-      })
-      it('supports multiple prefixes', async () => {
-        const { stdout } = await execa('node', [bin, 'FOOX', 'BARX'])
-        snapshot(stdout)
-      })
-      it('displays a message when no env variables are found', async () => {
+      const bin = resolve('bin', 'print-env.js')
+
+      const succesTestCases = [
+        ['displays usage when no arguments given', []],
+        ['prints sorted variables with values', ['FOOX']],
+        ['prints sorted variables without values when passed -e|--exists flag', ['-e', 'FOOX']],
+        ['supports multiple prefixes', ['FOOX', 'BARX']],
+        ['exits successfully if at least one variable is found', ['NOTFOUND', 'FOO']]
+      ]
+
+      const failTestCases = [
+        ['exits with 1 when no env variables are found', ['BAZX']],
+        ['exits with 1 when no env variables exist', ['--exists', 'BAZX']]
+      ]
+
+      succesTestCases.forEach(([successCase, args]) =>
+        it(successCase, async () => {
+          const { stdout, exitCode } = await execa(bin, args)
+          strictEqual(exitCode, 0, 'Expected successful exit')
+          snapshot(stdout)
+        })
+      )
+
+      failTestCases.forEach(([failCase, args]) => it(failCase, async () => {
         try {
-          await execa('node', [bin, 'BAZX'])
+          await execa(bin, args)
         } catch ({ stderr, exitCode }) {
-          snapshot(stderr.trim())
-          if (exitCode !== 1) {
-            throw new Error(
-              'Expected exit code to be 1 if no variables were found'
-            )
-          }
+          strictEqual(exitCode, 1, 'Expected exit code to be 1')
+          snapshot(stderr)
         }
-      })
-
-      it('exits with 1 if no env variables exist', async () => {
-        try {
-          await execa('node', [bin, '--exists', 'BAZX'])
-        } catch ({ exitCode }) {
-          if (exitCode !== 1) {
-            throw new Error('Expected exit code to be 1 if no variables exist')
-          }
-        }
-      })
-
-      it('exits with 0 if finds some variables from the list', async () => {
-        const { stdout, exitCode } = await execa('node', [bin, 'NONONO', 'FOO'])
-        if (exitCode !== 0) {
-          throw new Error('Expected successful exit')
-        }
-        snapshot('does not find NONONO, but finds FOO*', stdout)
-      })
+      }))
     })
   })
 })

